client/landing/gutenboarding/analytics.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { useEffect } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useOnUnmount } from 'landing/gutenboarding/hooks/use-on-unmount';
+import { StepNameType } from './path';
+
+export function recordLeaveStep( stepName: StepNameType ) {
+	recordTracksEvent( 'calypso_signup_step_leave', {
+		flow: 'gutenboarding',
+		step: stepName,
+	} );
+}
+
+export function recordEnterStep( stepName: StepNameType ) {
+	recordTracksEvent( 'calypso_signup_step_enter', {
+		flow: 'gutenboarding',
+		step: stepName,
+	} );
+}
+
+export function useTrackStep( stepName: StepNameType ) {
+	useOnUnmount( () => {
+		recordLeaveStep( stepName );
```

Comment:
This is nice a way of tracking when a user navigates away from a step! 

We'll probably have to pass some params to it as well (probably to be defined) 

For example, which design a user selects at the design step

Reply:
thanks, and good point that we will need params